src/components/Collect.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import Image from "next/image";
+import {
+  HaLoNoncePCDArgs,
+  HaLoNoncePCD,
+  HaLoNoncePCDPackage,
+} from "@pcd/halo-nonce-pcd";
+import { useEffect, useState } from "react";
+import { cardPubKeys } from "@/lib/cardPubKeys";
+import { saveSigmoji } from "@/lib/localStorage";
+import { useRouter } from "next/navigation";
+import { detectIncognito } from "detectincognitojs";
+import { Button } from "./ui/Button";
+import { Sigmoji } from "@/types";
+import { attestationText } from "@/lib/attestationData";
+import { useSigmojis } from "@/hooks/useSimojis";
+
+interface CollectProps {
+  args: HaLoNoncePCDArgs;
+}
+export default function Collect({ args }: CollectProps) {
+  const router = useRouter();
+
+  const [pcd, setPCD] = useState<HaLoNoncePCD | undefined>(undefined);
+  const [imageLink, setImageLink] = useState<string | undefined>(undefined);
+  const [alreadyCollected, setAlreadyCollected] = useState(false);
+  const { data: sigmojis = [] } = useSigmojis();
+
+  const alertIncognito = async () => {
+    const isIncognito = await detectIncognito();
+    if (isIncognito.isPrivate) {
+      alert(
+        "Please copy this link into a non-incognito tab in order to save your Sigmojis!"
+      );
+    }
+  };
+
+  useEffect(() => {
+    const generatePCD = async () => {
+      // make the main signature PCD
+      let producedPCD;
+      try {
+        producedPCD = await HaLoNoncePCDPackage.prove(args);
+      } catch (e) {
+        router.push("/");
+        return;
+      }
+      if (
+        producedPCD === undefined ||
+        !(await HaLoNoncePCDPackage.verify(producedPCD))
+      ) {
+        router.push("/");
+        return;
+      }
+
+      // make sure we haven't already collected this sigmoji if it isn't an event one
+      for (const sigmoji of sigmojis) {
+        if (
```

Comment:
What is this if statement for?

Reply:
Should add a comment explaining this

Reply:
Yep we should definitely remove this. And a lot of this logic should be factored out